diff --git a/libs/health.js b/libs/health.js
index 22bdcc6..e7f38ce 100644
--- a/libs/health.js
+++ b/libs/health.js
@@ -16,7 +16,7 @@ module.exports = function(s,config,lang,io){
                 k.cmd="ps -A -o %cpu | awk '{s+=$1} END {print s}'";
             break;
             case'linux':
-                k.cmd='top -b -n 2 | awk \'{IGNORECASE = 1} /^.?CPU/ {gsub("id,","100",$8); gsub("%","",$8); print 100-$8}\' | tail -n 1';
+                k.cmd='top -b -n 2 | awk \'{IGNORECASE = 1} /^.?Cpu/ {gsub("id,","100",$8); gsub("%","",$8); print 100-$8}\' | tail -n 1';
             break;
             case'freebsd':
                 k.cmd='vmstat 1 2 | awk \'END{print 100-$19}\''
