diff --git a/definitions/en_CA.js b/definitions/en_CA.js
index 15a1aa5..0ee9b6b 100644
--- a/definitions/en_CA.js
+++ b/definitions/en_CA.js
@@ -770,6 +770,10 @@ module.exports = function(s,config,lang){
                                     "name": "H.264 NVENC (NVIDIA HW Accel)",
                                     "value": "h264_nvenc"
                                  },
+                                 {
+                                    "name": "H.264 NVMPI (NVIDIA HW Accel)",
+                                    "value": "h264_nvmpi"
+                                 },
                                  {
                                     "name": "H.265 NVENC (NVIDIA HW Accel)",
                                     "value": "hevc_nvenc"
@@ -863,7 +867,7 @@ module.exports = function(s,config,lang){
                       "description": "How long each video segment should be, in minutes. Each segment will be drawn by the client through an m3u8 file. Shorter segments take less space.",
                       "default": "2",
                       "example": "",
-                      "form-group-class-pre-layer": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class-pre-layer": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "form-group-class": "h_st_input h_st_hls",
                       "possible": ""
                    },
@@ -873,7 +877,7 @@ module.exports = function(s,config,lang){
                       "description": "The number of segments maximum before deleting old segments automatically.",
                       "default": "2",
                       "example": "",
-                      "form-group-class-pre-layer": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class-pre-layer": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "form-group-class": "h_st_input h_st_hls",
                       "possible": ""
                    },
@@ -883,7 +887,7 @@ module.exports = function(s,config,lang){
                       "description": "Preset flag for certain video encoders. If you find your camera is crashing every few seconds : try leaving it blank.",
                       "default": "",
                       "example": "ultrafast",
-                      "form-group-class-pre-layer": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class-pre-layer": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "form-group-class": "h_st_input h_st_hls h_st_flv h_st_mp4 h_st_h265",
                       "possible": ""
                    },
@@ -893,7 +897,7 @@ module.exports = function(s,config,lang){
                       "description": "Low number means higher quality. Higher number means less quality.",
                       "default": "15",
                       "example": "1",
-                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "possible": "1-23"
                    },
                    {
@@ -902,7 +906,7 @@ module.exports = function(s,config,lang){
                       "description": "The speed in which frames are displayed to clients, in Frames Per Second. Be aware there is no default. This can lead to high bandwidth usage.",
                       "default": "",
                       "example": "1",
-                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "possible": ""
                    },
                    {
@@ -913,7 +917,7 @@ module.exports = function(s,config,lang){
                       "fieldType": "number",
                       "numberMin": "1",
                       "example": "640",
-                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "possible": ""
                    },
                    {
@@ -924,7 +928,7 @@ module.exports = function(s,config,lang){
                       "fieldType": "number",
                       "numberMin": "1",
                       "example": "480",
-                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "possible": ""
                    },
                    {
@@ -934,7 +938,7 @@ module.exports = function(s,config,lang){
                       "default": "",
                       "example": "",
                       "fieldType": "select",
-                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "possible": [
                            {
                               "name": lang["No Rotation"],
@@ -994,7 +998,7 @@ module.exports = function(s,config,lang){
                       "description": "Place FFMPEG video filters in this box to affect the streaming portion. No spaces.",
                       "default": "",
                       "example": "",
-                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                      "form-group-class": "h_st_input h_st_mjpeg h_st_b64 h_st_hls h_st_gif h_st_flv h_st_mp4 h_st_h265 h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                       "possible": ""
                   },
                    {
@@ -1041,7 +1045,7 @@ module.exports = function(s,config,lang){
                 "name": lang["Stream Timestamp"],
                 "color": "blue",
                 isAdvanced: true,
-                "section-class": "h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                "section-class": "h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                 "isSection": true,
                 "info": [
                     {
@@ -1130,7 +1134,7 @@ module.exports = function(s,config,lang){
                 "name": lang['Stream Watermark'],
                 "color": "blue",
                 isAdvanced: true,
-                "section-class": "h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_hevc_nvenc h_hls_v_no",
+                "section-class": "h_hls_v_input h_hls_v_libx264 h_hls_v_libx265 h_hls_v_h264_nvenc h_hls_v_h264_nvmpi h_hls_v_hevc_nvenc h_hls_v_no",
                 "isSection": true,
                 "info": [
                     {
@@ -1376,6 +1380,10 @@ module.exports = function(s,config,lang){
                                     "name": "H.264 NVENC (NVIDIA HW Accel)",
                                     "value": "h264_nvenc"
                                  },
+                                 {
+                                    "name": "H.264 NVMPI (NVIDIA HW Accel)",
+                                    "value": "h264_nvmpi"
+                                 },
                                  {
                                     "name": "H.265 NVENC (NVIDIA HW Accel)",
                                     "value": "hevc_nvenc"
@@ -1423,7 +1431,7 @@ module.exports = function(s,config,lang){
                       "description": "Low number means higher quality. Higher number means less quality.",
                       "default": "15",
                       "example": "1",
-                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                       "possible": "1-23"
                    },
                    {
@@ -1432,7 +1440,7 @@ module.exports = function(s,config,lang){
                       "description": "Preset flag for certain video encoders. If you find your camera is crashing every few seconds : try leaving it blank.",
                       "default": "",
                       "example": "ultrafast",
-                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                       "possible": ""
                    },
                    {
@@ -1497,7 +1505,7 @@ module.exports = function(s,config,lang){
                       "description": "The speed in which frames are recorded to files, Frames Per Second. Be aware there is no default. This can lead to large files. Best to set this camera-side.",
                       "default": "",
                       "example": "2",
-                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                       "possible": ""
                    },
                    {
@@ -1506,7 +1514,7 @@ module.exports = function(s,config,lang){
                       "description": "Height of the stream image.",
                       "default": "",
                       "example": "720",
-                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                       "possible": ""
                    },
                    {
@@ -1515,7 +1523,7 @@ module.exports = function(s,config,lang){
                       "description": "Width of the stream image.",
                       "default": "",
                       "example": "1280",
-                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                       "possible": ""
                    },
                    {
@@ -1535,7 +1543,7 @@ module.exports = function(s,config,lang){
                       "example": "",
                       "selector": "h_vc",
                       "fieldType": "select",
-                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                       "possible": [
                            {
                               "name": lang["No Rotation"],
@@ -1569,7 +1577,7 @@ module.exports = function(s,config,lang){
                       "description": "Place FFMPEG video filters in this box to affect the recording portion. No spaces.",
                       "default": "",
                       "example": "",
-                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                      "form-group-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                       "possible": ""
                    }
                 ]
@@ -1579,7 +1587,7 @@ module.exports = function(s,config,lang){
                 "name": lang['Recording Timestamp'],
                 "color": "red",
                 isAdvanced: true,
-                "section-pre-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                "section-pre-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                 "section-class": "h_m_input h_m_record h_m_idle",
                 "isSection": true,
                 "info": [
@@ -1669,7 +1677,7 @@ module.exports = function(s,config,lang){
                 "name": lang['Recording Watermark'],
                 "color": "red",
                 isAdvanced: true,
-                "section-pre-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
+                "section-pre-class": "h_vc_input h_vc_libvpx h_vc_libvpx-vp9 h_vc_libx264 h_vc_libx265 h_vc_hevc_nvenc h_vc_h264_nvenc h_vc_h264_nvmpi h_vc_h264_vaapi h_vc_hevc_vaapi h_vc_h264_qsv h_vc_hevc_qsv h_vc_mpeg2_qsv h_vc_default h_vc_none",
                 "section-class": "h_m_input h_m_record h_m_idle",
                 "isSection": true,
                 "info": [
diff --git a/languages/en_CA.json b/languages/en_CA.json
index 3ad5cf3..3635b82 100644
--- a/languages/en_CA.json
+++ b/languages/en_CA.json
@@ -873,6 +873,7 @@
    "h264_omx": "H.264 openMAX (Raspberry Pi)",
    "h264_vaapi": "H.264 VA-API (Intel HW Accel)",
    "h264_nvenc": "H.264 NVENC (NVIDIA HW Accel)",
+   "h264_nvmpi": "H.264 NVMPI (NVIDIA HW Accel)",
    "hevc_nvenc": "H.265 NVENC (NVIDIA HW Accel)",
    "hevc_vaapi": "H.265 VA-API (Intel HW Accel)",
    "vp8_cuvid": "VP8 NVENC (NVIDIA HW Accel)",
